internal/cmd/run.go

```diff
@@ -504,6 +498,22 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) (err error) {
 	return nil
 }
 
+func getSummaryMode(runtimeOptions lib.RuntimeOptions, logger logrus.FieldLogger) (summary.Mode, bool) {
+	if runtimeOptions.NoSummary.Bool {
+		return summary.ModeDisabled, false
+	}
+
+	sm, err := summary.ValidateMode(runtimeOptions.SummaryMode.String)
+	if err != nil {
+		logger.WithError(err).Warnf(
+			"invalid summary mode %q, falling back to \"compact\" (default)",
+			runtimeOptions.SummaryMode.String,
+		)
+	}
```

Comment:
This seems like a thing that should abort the execution?

Reply:
I have mixed feelings here, but yeah, maybe.

In any case this is not something new, see https://github.com/grafana/k6/blob/master/internal/cmd/run.go#L217.
So, if you suggest this, I could agree, but then I'd leave for a separate PR.

Reply:
So, the fact is that this should never happen, unless we eventually introduce a "bug" that drops the early validation.
In such case, what would happen here? Again, throw an error? Panic? I can leave a comment explaining why.